lib/helpers/wp-cli-db.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * This file is sourced in wp-config.php so we can limit the critical path to the specific `wp db` command.
+ */
+
+if ( defined( 'WPINC' ) ) {
+	// Don't do anything when WordPress is loaded.
+	return;
+}
+
+if ( ! class_exists( 'WP_CLI' ) ) {
+	return;
+}
+
+WP_CLI::add_hook( 'before_run_command', function ( $command ) {
+	global $db_servers;
+
+	if ( ! ( isset( $command[0] ) && 'db' === $command[0] ) ) {
+		// Don't do anything for any command other than `db`
+		return;
+	}
+
+	if ( ! is_array( $db_servers ) ) {
+		echo "Error: No database servers are configured for this environment.\n";
+		exit( 10 );
```

Comment:
Why the `10` specifically?

Reply:
No specific reason other than to have a non-zero exit code (to signify an error). We should decide which code(s) we want to support for each error if we want to differentiate (or just use `1` if not).